### PR TITLE
Feat amm

### PR DESCRIPTION
**Pull request by BruceChar**
new router mapping

### CODE REVIEW

File: packages/app/src/controller/health.ts [hunk 1]
```diff
@@ -5,11 +5,9 @@ import { RedisService } from '../service/redis'
 
 export async function healthCheck(ctx: Context, next: Next) {
```
Comment (crystalynliu) [inline packages/app/src/controller/health.ts:6]: what's the health check for?

File: packages/app/src/controller/mm/market.ts [hunk 2]
```diff
@@ -1,114 +1,89 @@
...
+import { RedisService } from '../../service/redis'
+import { LendingAssetConfigure } from '../../models'
 
 const log = getAppLogger('controller-mm-market')
 
-type DateItem = {
-    date: string
-    assets: string[]
+export async function GetAllMarketMetadatas(ctx: Context, next: Next) {
+    const re = await RedisService.getMarketMetaList()
```
Comment (crystalynliu) [inline packages/app/src/controller/mm/market.ts:10]: If there is no MM list from redis, what's the next step?

File: packages/app/src/service/redis.ts [hunk 3]
```diff
@@ -49,18 +50,121 @@ export const userRd = userRedis.getClient()
...
+    static async dumpAssetMetaCache() {
         const assetIds = await ApiService.getAssets()
+        let metas = []
         for (let assetId of assetIds) {
             const meta = await ApiService.getAssetMeta(assetId)
-            await Promise.all([this.setToken(assetId, meta.symbol), this.setDecimals(assetId, meta.decimals)])
+            metas.push(this.setAssetMeta(assetId, meta))
+        }
+        metas.push(
+            this.setAssetMeta(0, {
```
Comment (crystalynliu) [inline packages/app/src/service/redis.ts:69]: why is It hardcode?

File: packages/app/src/service/subql/lending.ts [hunk 4]
```diff
@@ -85,17 +72,16 @@ async function positionHandler(node: ActionNode) {
...
+        const day = blockTimeToDayValue(node.timestamp)
 
         await getConnection()
             .getRepository(LendingPosition)
             .save({
                 id: `${node.address}-${node.assetId}-${day}`,
                 address: node.address,
-                symbol,
-                decimals,
+                symbol: meta.symbol,
```
Comment (crystalynliu) [inline packages/app/src/service/subql/lending.ts:83]: I think we discussed we don't need to return symbol and decimal for other endpoints except for Asset?

File: packages/app/src/controller/health.ts (hunk 1 above)
Comment (yz89) [reply packages/app/src/controller/health.ts:6]: I think it's for monitoring.

Comment (crystalynliu) [reply packages/app/src/controller/health.ts:6]: Looks weird, we can discuss the monitor later. What does it check? Service available or something else?

**Review by yz89 — APPROVED** [review]

File: packages/app/src/controller/health.ts (hunk 1 above)
Comment (yz89) [reply packages/app/src/controller/health.ts:6]: Maybe check if the service is alive. @BruceChar

Comment (BruceChar) [reply packages/app/src/controller/health.ts:6]: Yiyang deploy need this for heartbeat

File: packages/app/src/controller/mm/market.ts (hunk 2 above)
Comment (BruceChar) [reply packages/app/src/controller/mm/market.ts:10]: Maybe need to expose an error response. If the list is Empty, network would be fail, and the data will be refetch in the next day!

File: packages/app/src/service/redis.ts (hunk 3 above)
Comment (BruceChar) [reply packages/app/src/service/redis.ts:69]: The DOT USDT and HKO can not be fetched from the chain state.
And there is no metadata info for these assets.

File: packages/app/src/controller/health.ts (hunk 1 above)
Comment (crystalynliu) [reply packages/app/src/controller/health.ts:6]: If it's used by loadbalance, I think we can just simply return the 200, why does there are some logics inside it? Maybe refer to the auction-service health check endpoint.

https://github.com/parallel-finance/auction-service-v2/blob/main/src/router/heartbeat.ts

File: packages/app/src/service/subql/lending.ts (hunk 4 above)
Comment (BruceChar) [reply packages/app/src/service/subql/lending.ts:84]: Forgot to remove the fields. Some schema maybe can keep the fields for convinience, or remove all the fields. In some scope, e.g. Market, the market Id is subtle different form asset id, but they are the same thing. Do we need to keep the symbol field for the market metadata?

File: packages/app/src/service/redis.ts (hunk 3 above)
Comment (crystalynliu) [reply packages/app/src/service/redis.ts:69]: I don't think so, all the asset metadata should be queried from the blockchain. is it? @yz89 
And even if we cannot get it, it's not a good idea put it in Redis, we can put it in RDS

File: packages/app/src/service/subql/lending.ts (hunk 4 above)
Comment (crystalynliu) [reply packages/app/src/service/subql/lending.ts:84]: What's the difference between `assetId` and `marketId`?

Comment (BruceChar) [reply packages/app/src/service/subql/lending.ts:84]: Some asset dont in the market list.

Comment (crystalynliu) [reply packages/app/src/service/subql/lending.ts:84]: I think we have some gap here, we have an asset endpoint, which should include all the assets? Why do we care if there are some assets not in the market list. Maybe we need to clarify it in the slack channel first.

Comment (BruceChar) [reply packages/app/src/service/subql/lending.ts]: eh.  it's different about 'api.query.assets.asset.keys()' and ]api.query.loans.markets.keys()' , maybe we can check with Andrew or Cj

**Pull request merged**